gobblin-compaction/src/main/java/gobblin/compaction/dataset/TimeBasedSubDirDatasetsFinder.java

```diff
@@ -102,55 +107,84 @@ private String getDatasetName(String path, String basePath) {
-          Path outputLatePath = new Path(this.destDir, new Path(datasetName, this.destLateSubDir));
-          Path outputTmpPath = new Path(this.tmpOutputDir, new Path(datasetName, this.destSubDir));
-          double priority = this.getDatasetPriority(datasetName);
-
-          String folderStructure = getFolderStructure();
-          for (FileStatus status : this.fs.globStatus(new Path(inputPath, folderStructure))) {
-            Path jobInputPath = status.getPath();
-            DateTime folderTime = null;
-            try {
-              folderTime = getFolderTime(jobInputPath, inputPath);
-            } catch (RuntimeException e) {
-              log.warn(jobInputPath + " is not a valid folder. Will be skipped.");
-              continue;
-            }
-
-            if (folderWithinAllowedPeriod(jobInputPath, folderTime)) {
-              Path jobInputLatePath = appendFolderTime(inputLatePath, folderTime);
-              Path jobOutputPath = appendFolderTime(outputPath, folderTime);
-              Path jobOutputLatePath = appendFolderTime(outputLatePath, folderTime);
-              Path jobOutputTmpPath = appendFolderTime(outputTmpPath, folderTime);
-
-              Dataset timeBasedDataset = new Dataset.Builder().withPriority(priority)
-                  .withDatasetName(datasetName)
-                  .withInputPath(this.recompactDatasets ? jobOutputPath : jobInputPath)
-                  .withInputLatePath(this.recompactDatasets ? jobOutputLatePath : jobInputLatePath)
-                  .withOutputPath(jobOutputPath).withOutputLatePath(jobOutputLatePath)
-                  .withOutputTmpPath(jobOutputTmpPath).build();
-              // Stores the extra information for timeBasedDataset
-              timeBasedDataset.setJobProp(MRCompactor.COMPACTION_JOB_DEST_PARTITION,
-                  folderTime.toString(this.timeFormatter));
-              timeBasedDataset.setJobProp(MRCompactor.COMPACTION_INPUT_PATH_TIME, folderTime.getMillis());
-              datasets.add(timeBasedDataset);
-            }
+
+          buildDatasetForInput(datasetsBuilderMap, datasetName, inputPath, inputLatePath);
+
+          if (this.alternateInputDir.isPresent()) {
+            Path alternateInputPath = new Path(this.alternateInputDir.get(), new Path(datasetName, this.inputSubDir));
+            Path alternateInputLatePath = new Path(this.alternateInputDir.get(), new Path(datasetName, this.inputLateSubDir));
+
+            buildDatasetForInput(datasetsBuilderMap, datasetName, alternateInputPath, alternateInputLatePath);
           }
         }
       }
     }
-    return datasets;
+
+    return Sets.newHashSet(Collections2.transform(datasetsBuilderMap.values(), new Function<Builder, Dataset>() {
+
+      @Override
+      public Dataset apply(Builder b) {
+        return b.build();
+      }
+    }));
+
+  }
+
+  private void buildDatasetForInput(Map<DateTime, Dataset.Builder> datasetsBuilderMap, String datasetName,
```

Comment:
Removed this code. Will be extending this finder and creating a li-finder. li-finder will have tests